plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/CrowdFunding.hs

```diff
@@ -117,48 +124,48 @@ contributionScript cmp  = ValidatorScript $
 campaignAddress :: Campaign -> Ledger.Address
 campaignAddress = Ledger.scriptAddress . contributionScript
 
--- | Contribute funds to the campaign (contributor)
---
-contribute :: MonadWallet m => Slot -> Value -> Slot -> Wallet -> Value -> m ()
-contribute deadline target collectionDeadline ownerWallet contribution = do
-    let cmp = mkCampaign deadline target collectionDeadline ownerWallet
-    ownPK <- ownPubKey
-    let ds = DataScript (Ledger.lifted ownPK)
-        range = W.interval 1 (campaignDeadline cmp)
-    tx <- payToScript range (campaignAddress cmp) contribution ds
-    logMsg "Submitted contribution"
-
-    register (refundTrigger contribution cmp) (refundHandler (Ledger.hashTx tx) cmp)
-    logMsg "Registered refund trigger"
-
--- | Register a [[EventHandler]] to collect all the funds of a campaign
---
-scheduleCollection :: MonadWallet m => Slot -> Value -> Slot -> Wallet -> m ()
-scheduleCollection deadline target collectionDeadline ownerWallet = do
-    let cmp = mkCampaign deadline target collectionDeadline ownerWallet
-    register (collectFundsTrigger cmp) (EventHandler (\_ -> do
-        logMsg "Collecting funds"
-        let redeemerScript = Ledger.RedeemerScript (Ledger.lifted Collect)
-            range = collectionRange cmp
-        collectFromScript range (contributionScript cmp) redeemerScript))
-
--- | An event trigger that fires when a refund of campaign contributions can be claimed
-refundTrigger :: Value -> Campaign -> EventTrigger
-refundTrigger vl c = andT
-    (fundsAtAddressGeqT (campaignAddress c) vl)
-    (slotRangeT (refundRange c))
-
--- | An event trigger that fires when the funds for a campaign can be collected
-collectFundsTrigger :: Campaign -> EventTrigger
-collectFundsTrigger c = andT
-    (fundsAtAddressGeqT (campaignAddress c) (campaignTarget c))
-    (slotRangeT (collectionRange c))
-
--- | Claim a refund of our campaign contribution
-refundHandler :: MonadWallet m => TxId -> Campaign -> EventHandler m
-refundHandler txid cmp = EventHandler (\_ -> do
-    logMsg "Claiming refund"
-    let validatorScript = contributionScript cmp
-        redeemerScript  = Ledger.RedeemerScript (Ledger.lifted Refund)
-
-    collectFromScriptTxn (refundRange cmp) validatorScript redeemerScript txid)
+crowdfunding :: ContractActions r => Contract r ()
+crowdfunding = contribute theCampaign <|> scheduleCollection theCampaign
+
+theCampaign :: Campaign
+theCampaign = Campaign
```

Comment:
Hm, maybe parameterize `crowdfunding` so we don't have to commit to this in the example file?